lib/thrift/generator/client.ex

```diff
@@ -0,0 +1,270 @@
+defmodule Thrift.Generator.ClientGenerator do
+  defmodule Framed do
+    alias Thrift.Generator.Service
+    alias Thrift.Parser.Models.Function
+
+    def generate(service_module, service) do
+      functions = service.functions
+      |> Map.values
+      |> Enum.map(&generate_function(service_module, &1))
+
+      quote do
+        defmodule Client.Framed do
+          alias Thrift.Protocols.Binary, as: BinaryBase
+          alias Thrift.Exceptions.TApplicationException
+
+          defmodule State do
+            defstruct host: nil, port: nil, tcp_opts: nil, timeout: 5000, sock: nil
+          end
+          use Connection
+
+          def init({host, port, tcp_opts, timeout}) do
+            s = %State{host: host,
+                       port: port,
+                       tcp_opts: tcp_opts,
+                       timeout: timeout,
+                       sock: nil}
+
+            {:connect, :init, s}
+          end
+
+          def start_link(host, port, tcp_opts, timeout \\ 5000) do
+            Connection.start_link(__MODULE__, {host, port, tcp_opts, timeout})
+          end
+
+          def send(conn, data), do: Connection.call(conn, {:send, data})
+
+          def recv(conn, bytes, timeout \\ 5000) do
+            Connection.call(conn, {:recv, bytes, timeout})
+          end
```

Comment:
Can we inline this send/recv? Makes me uneasy exposing them alongside the user's generated methods. In fact, could the connection behavior stuff go into a separate module from the generated stuff?